### PR TITLE
Rearchitecture of TypeScript files

### PR DESCRIPTION
**Pull request by johnpaliotta**
Purpose is to organize the commands and calls to clicast, vpython, and vcastqt in a logical way.
This was a pre-requisite for implementing an environment server.

### CODE REVIEW

File: src/testPane.ts [hunk 1]
```diff
@@ -314,12 +305,11 @@ function getEnvironmentList(baseDirectory: string): string[] {
...
 
 // These variables are used in the package.json to control the display of context menu items
-// Search for 'vectorcastTestExplorer.vcastEnviroList | vcastHasCodedTestsList' in package.json 
+// Search for 'vectorcastTestExplorer.vcastEnviroList | vcastHasCodedTestsList' in package.json
 // to see where we reference them
 // this list in a "when" clause
-export var vcastEnviroList: string[] = [];
-export var vcastHasCodedTestsList: string[] = [];
-
+let vcastEnviroList: string[] = [];
```
Comment (Zbigor) [inline src/testPane.ts:311]: no need for export anymore (we're not using these in other files)?

File: src/vcastAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+
+const path = require("path");
+
+// ------------------------------------------------------------------------------------
+// Direct clicast Calls
+// ------------------------------------------------------------------------------------
+
+// Check License
+export function vcastLicenseOK(): boolean {
+  const commandToRun = `${clicastCommandToUse} tools has_license`;
```
Comment (Zbigor) [inline src/vcastAdapter.ts:42]: Should we use clicast interface?

**Review by Zbigor — APPROVED** [review]
After going through the diffs, I mapped out where things got moved and how they changed. Changes look good

File: python/clicastInterface.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+
+
+enviroNameRegex = "-e([^\s]*)"
+
+
+def getEnviroPathFromCommand(command):
+
+    # TBD in the future we will change the caller to pass in the environment path
+    # No error handling because the caller will guarantee that we have a valid command
+    match = re.search(enviroNameRegex, command)
```
Comment (aytey) [inline python/clicastInterface.py:22]: Who generates what's passed to this? It is valid to do `-e<space><env name>`.

The regex could be:

```py
enviroNameRegex = "-e\s*([^\s]*)"
```

File: src/testPane.ts (hunk 1 above)
Comment (johnpaliotta) [reply src/testPane.ts:311]: The export was removed in this change :)

**Comment by johnpaliotta** [comment]
> After going through the diffs, I mapped out where things got moved and how they changed. Changes look good

Thanks!

**Pull request merged**